python/rapidsmpf/rapidsmpf/integrations/single.py

```diff
@@ -0,0 +1,222 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+"""Shuffler integration for single-worker pylibcudf execution."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, cast
+
+from rapidsmpf.communicator.single import new_communicator
+from rapidsmpf.config import Options
+from rapidsmpf.integrations.core import (
+    WorkerContext,
+    extract_partition,
+    get_new_shuffle_id,
+    get_shuffler,
+    insert_partition,
+    rmpf_worker_setup,
+)
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    from rapidsmpf.integrations.core import ShufflerIntegration
+
+
+# Local single-worker context
+class _SingleWorker:
+    """Mutable single-worker utility class."""
+
+    context: WorkerContext | None = None
+
+
+_single_rapidsmpf_worker: _SingleWorker = _SingleWorker()
+
+
+def _get_single_worker_context(worker: _SingleWorker | None = None) -> WorkerContext:
+    """
+    Retrieve the single :class:`rapidsmpf.integrations.core.WorkerContext`.
+
+    If the worker context does not already exist on the worker, it
+    will be created and attached to the ``_single_rapidsmpf_worker``
+    singleton.
+
+    Returns
+    -------
+    The existing or newly initialized worker context.
+    """
+    with WorkerContext.lock:
+        worker = worker or _single_rapidsmpf_worker
+        if worker.context is None:
+            worker.context = WorkerContext()
+        return cast("WorkerContext", worker.context)
+
+
+def setup_single_worker(options: Options = Options()) -> None:
```

Comment:
Similarly here, we're in the `single` module, do we really need `single` in the name of these functions?